### ISSUE TITLE
A question on GroupNorm in multi-GPU setting

### ISSUE THREAD

**Issue by Remember2018** [tensorpack/tensorpack#1353]
Hi Yuxin, in the `GroupNorm` of `FasterRCNN` module, the `GroupNorm` is implemented based on the `batch normalization`. In multi-GPU setting, do we need to use the SyncBN mechanism in the GroupNorm? Thanks.

    @layer_register(log_shape=True)
    def GroupNorm(x, group=32, gamma_initializer=tf.constant_initializer(1.),
                  data_format='channels_first', **other_params):
        """
        More code that reproduces the paper can be found at https://github.com/ppwwyyxx/GroupNorm-reproduce/.
        """
        shape = x.get_shape().as_list()
        ndims = len(shape)
        assert ndims == 4, shape
        if data_format == 'channels_first':
            chan = shape[1]
        else:
            chan = shape[3]
        assert chan % group == 0, chan
        group_size = chan // group
    
        orig_shape = tf.shape(x)
        if data_format == 'channels_first':
            h, w = orig_shape[2], orig_shape[3]
        else:
            h, w = orig_shape[1], orig_shape[2]
    
        if data_format == 'channels_first':
            x = tf.reshape(x, tf.stack([-1, group, group_size, h, w]))
            mean, var = tf.nn.moments(x, [2, 3, 4], keep_dims=True)
            new_shape = [1, group, group_size, 1, 1]
        else:
            x = tf.reshape(x, tf.stack([-1, h, w, group, group_size]))
            mean, var = tf.nn.moments(x, [1, 2, 4], keep_dims=True)
            new_shape = [1, 1, 1, group, group_size]
    
        beta = tf.get_variable('beta', [chan], initializer=tf.constant_initializer())
        beta = tf.reshape(beta, new_shape)
    
        gamma = tf.get_variable('gamma', [chan], initializer=gamma_initializer)
        gamma = tf.reshape(gamma, new_shape)
    
        out = tf.nn.batch_normalization(x, mean, var, beta, gamma, 1e-5, name='output')
        return tf.reshape(out, orig_shape, name='output')

**Comment by ppwwyyxx** [comment]
No. `tf.nn.batch_normalization` is not batch normalization.

**Comment by Remember2018** [comment]
@ppwwyyxx Thanks a lot, Yuxin. Sorry for my misunderstanding for `GroupNorm` and `tf.nn.batch_normalization`. I have read your paper on the `GroupNorm`. `GroupNorm` is indenpendent of the batch size. So no synchronization is needed.
